lightly_studio/src/lightly_studio/models/video.py

```diff
@@ -0,0 +1,105 @@
+"""This module defines the Video and VideoFrame model for the application."""
+
+from typing import TYPE_CHECKING, List
+from uuid import UUID
+
+from sqlalchemy.orm import Mapped
+from sqlmodel import Field, Relationship, SQLModel
+
+if TYPE_CHECKING:
+    from lightly_studio.models.annotation.annotation_base import (
+        AnnotationBaseTable,
+    )
+    from lightly_studio.models.sample import SampleTable, SampleView
+
+else:
+    AnnotationBaseTable = object
+    SampleTable = object
+
+
+class VideoBase(SQLModel):
+    """Base class for the Video model."""
+
+    """The width of the video in pixels."""
+    width: int
+
+    """The height of the video in pixels."""
+    height: int
+
+    """The duration of the video in seconds."""
+    duration: float
+
+    """The fps of the video."""
+    fps: float
+
+    """The file name of the video."""
+    file_name: str
+
+    """The path of the video."""
+    file_path_abs: str = Field(default=None)
+
+
+class VideoCreate(VideoBase):
+    """Video class when inserting."""
+
+
+class VideoTable(VideoBase, table=True):
+    """This class defines the Video ORM table."""
+
+    __tablename__ = "video"
+    sample_id: UUID = Field(foreign_key="sample.sample_id", primary_key=True)
+    frames: Mapped[List["VideoFrameTable"]] = Relationship(back_populates="video")
+    sample: Mapped["SampleTable"] = Relationship()
+
+
+class VideoView(SQLModel):
```

Comment:
Optional: I'm not sure why we don't use VideoBase here (I know this has been copied from image.py). Can you try that? Similar for VideoFrameView.